From 76aa8ee75865c5f879a322eae09dcd37a103d7b2 Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <skardach@marvell.com>
Date: Mon, 10 Feb 2020 21:46:46 +0100
Subject: [PATCH 476/767] octeontx2-dpi: fix compilation warnings

commit e6391ed0270ef8dc153d36bcd90ec7db71d1b860 from
git@git.assembla.com:cavium/WindRiver.linux.git

Fix symbol locality and missing parameter documentation in dpi_init().

Change-Id: I6520e7df9d8a0db686333ed3cd2926420329efa0
Signed-off-by: Stanislaw Kardach <skardach@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/23185
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Subrahmanyam Nilla <snilla@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-dpi/dpi.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-dpi/dpi.c b/drivers/soc/marvell/octeontx2-dpi/dpi.c
index 1be37f3b5d50..9896a4488b40 100644
--- a/drivers/soc/marvell/octeontx2-dpi/dpi.c
+++ b/drivers/soc/marvell/octeontx2-dpi/dpi.c
@@ -46,7 +46,7 @@ static int dpi_dma_engine_get_num(void)
 	return DPI_MAX_ENGINES;
 }
 
-int dpi_queue_init(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
+static int dpi_queue_init(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
 {
 	int engine = 0;
 	int queue = vf;
@@ -87,7 +87,7 @@ int dpi_queue_init(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
 	return 0;
 }
 
-int dpi_queue_fini(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
+static int dpi_queue_fini(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
 {
 	u64 reg = 0ULL;
 	int engine = 0;
@@ -120,9 +120,10 @@ int dpi_queue_fini(struct dpipf *dpi, struct dpipf_vf *dpivf, u8 vf)
 /**
  * Global initialization of DPI
  *
+ * @dpi: DPI device context structure
  * @return Zero on success, negative on failure
  */
-int dpi_init(struct dpipf *dpi)
+static int dpi_init(struct dpipf *dpi)
 {
 	int engine = 0;
 	u64 reg = 0ULL;
@@ -154,7 +155,7 @@ int dpi_init(struct dpipf *dpi)
 	return 0;
 }
 
-int dpi_fini(struct dpipf *dpi)
+static int dpi_fini(struct dpipf *dpi)
 {
 	int engine = 0;
 	u64 reg = 0ULL;
@@ -172,7 +173,7 @@ int dpi_fini(struct dpipf *dpi)
 	return 0;
 }
 
-int dpi_queue_reset(struct dpipf *dpi, u16 queue)
+static int dpi_queue_reset(struct dpipf *dpi, u16 queue)
 {
 	/* TODO: add support */
 	return 0;
-- 
2.31.1

